src/plugins/owner/SingleOwnerPlugin.sol

```diff
@@ -86,7 +85,13 @@ contract SingleOwnerPlugin is ISingleOwnerPlugin, BasePlugin {
     {
         if (functionId == uint8(FunctionId.VALIDATION_OWNER)) {
             // Validate that the sender is the owner of the account or self.
-            if (sender != _owners[msg.sender] && sender != msg.sender) {
+            if (
+                sender != msg.sender
+                    && (
+                        _owners[msg.sender].data.length == 0
+                            || sender != abi.decode(_owners[msg.sender].data, (address))
```

Comment:
This is not guaranteed to have this abi encoding though, right? For example for WebAuthnValidator has a different encoding to ECDSA one which is assumed here